### PR TITLE
Modules updates

### PR DESCRIPTION
**Pull request by RGates94**
This PR fixes various issues with the module system, and partially ports arb_os to use the system.

### CODE REVIEW

File: arb_os/decompression.mini [hunk 1]
```diff
@@ -342,3 +328,7 @@ impure func decompressTx(
     ));
 }
 
+func dumbHack() {
```
Comment (edfelten) [inline arb_os/decompression.mini:331]: Add a comment explaining why this is here.

File: src/compile/ast.rs [hunk 2]
```diff
@@ -12,6 +12,9 @@ use crate::pos::Location;
 use crate::stringtable::StringId;
 use crate::uint256::Uint256;
 use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+
+pub type TypeTree = HashMap<(Vec<String>, usize), Type>;
```
Comment (edfelten) [inline src/compile/ast.rs:17]: Add a comment explaining what a `TypeTree` represents.

**Review by edfelten — CHANGES_REQUESTED** [review]
Looks good.  Suggest adding some comments.

File: arb_os/decompression.mini (hunk 1 above)
Comment (RGates94) [reply arb_os/decompression.mini:333]: Added.

File: src/compile/ast.rs (hunk 2 above)
Comment (RGates94) [reply src/compile/ast.rs:19]: Added.

**Review by edfelten — APPROVED** [review]
Looks good.

**Pull request merged**